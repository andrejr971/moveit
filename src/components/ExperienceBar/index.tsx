import { useChallenge } from 'hooks/challenge'
import { Container } from './styles'

const ExperienceBar = () => {
  const { currentExperience, experienceToNextLevel } = useChallenge()

  const porcentToNextLevel =
    Math.round(currentExperience * 100) / experienceToNextLevel

  return (
    <Container>
      <span>0 xp</span>
      <div>
        <div style={{ width: `${porcentToNextLevel}%` }} />

        <span
          className="current-experience"
          style={{ left: `${porcentToNextLevel}%` }}
        >
          {currentExperience} xp
        </span>
      </div>
      <span>{experienceToNextLevel} xp</span>
    </Container>
  )
}

export default ExperienceBar
