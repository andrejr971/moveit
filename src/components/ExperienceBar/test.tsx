import { render, screen } from '@testing-library/react'

import ExperienceBar from '.'

describe('<ExperienceBar />', () => {
  it('should render the span', () => {
    const { container } = render(<ExperienceBar />)

    expect(screen.findAllByText('0 xp'))

    expect(container.firstChild).toMatchSnapshot()
  })
})
