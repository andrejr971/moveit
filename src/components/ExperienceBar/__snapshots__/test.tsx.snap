// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExperienceBar /> should render the span 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 span {
  font-size: 1rem;
}

.c0 > div {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 4px;
  border-radius: 4px;
  background: var(--gray-line);
  margin: 0 1.5rem;
  position: relative;
}

.c0 > div > div {
  height: 4px;
  border-radius: 4px;
  background: var(--green);
}

.c0 span.current-experience {
  position: absolute;
  top: 12px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

<header
  class="c0"
>
  <span>
    0 xp
  </span>
  <div>
    <div
      style="width: 50%;"
    />
    <span
      class="current-experience"
      style="left: 50%;"
    >
      300 xp
    </span>
  </div>
  <span>
    600 xp
  </span>
</header>
`;
